rs/types/types/src/exhaustive.rs

```diff
@@ -375,15 +382,45 @@ impl ExhaustiveSet for SchnorrKeyId {
     }
 }
 
+impl ExhaustiveSet for VetKdCurve {
+    fn exhaustive_set<R: RngCore + CryptoRng>(_: &mut R) -> Vec<Self> {
+        VetKdCurve::iter().collect()
+    }
+}
+
+impl ExhaustiveSet for VetKdKeyId {
+    fn exhaustive_set<R: RngCore + CryptoRng>(rng: &mut R) -> Vec<Self> {
+        <(VetKdCurve, String)>::exhaustive_set(rng)
+            .into_iter()
+            .map(|elem| Self {
+                curve: elem.0,
+                name: elem.1,
+            })
+            .collect()
+    }
+}
+
 impl ExhaustiveSet for MasterPublicKeyId {
     fn exhaustive_set<R: RngCore + CryptoRng>(rng: &mut R) -> Vec<Self> {
+        assert_eq!(MasterPublicKeyId::COUNT, 3);
         let ecdsa_key_ids = EcdsaKeyId::exhaustive_set(rng);
         let schnorr_key_ids = SchnorrKeyId::exhaustive_set(rng);
+        let vetkd_key_ids = VetKdKeyId::exhaustive_set(rng);
 
         ecdsa_key_ids
             .into_iter()
             .map(MasterPublicKeyId::Ecdsa)
             .chain(schnorr_key_ids.into_iter().map(MasterPublicKeyId::Schnorr))
+            .chain(vetkd_key_ids.into_iter().map(MasterPublicKeyId::VetKd))
+            .collect()
+    }
+}
+
+impl ExhaustiveSet for IDkgMasterPublicKeyId {
+    fn exhaustive_set<R: RngCore + CryptoRng>(rng: &mut R) -> Vec<Self> {
+        MasterPublicKeyId::exhaustive_set(rng)
+            .into_iter()
+            .flat_map(IDkgMasterPublicKeyId::try_from)
```

Comment:
NIT:
I know that this is much shorter and concise, but I find the fact that `Result` implements `IntoIterator` and that you can use that to use `flat_map` to filter out errors kinda cursed.

Even though it is quite a bit longer, I would use a `filter_map` and match on the results.